packages-backend/express/src/auth.ts

```diff
@@ -104,15 +104,15 @@ const getConfiguredDefaultIssuer = (options: TSessionMiddlewareOptions) => {
 // Construct the audience from the given option + the request path.
 // If the request path is `/`, do not append it to the audience, otherwise
 // the token validation might fail because of mismatching audiences.
-const getConfiguredAudience = (
+export const getConfiguredAudience = (
   options: TSessionMiddlewareOptions,
   requestPath: string
 ) => {
   const url = new URL(options.audience);
   if (requestPath !== '/') {
-    url.pathname = requestPath;
+    return `${url.origin}${requestPath}${url.search}`;
   }
-  return url.toString();
+  return `${url.origin}${url.search}`;
```

Comment:
I'm actually considering to completely remove the search params from the `audience` in the first place (when we forward the request and sign the token).

So we should fix this in both places (MC API and here).

Reply:
that is actually a good call! let's go for it! 🙌